posts/data-licensing.qmd

```diff
@@ -0,0 +1,61 @@
+---
+title: Dual licensing R packages with code and data
+date: 2024-09-23
+author:
+  - name: "Joshua Lambert"
+    orcid: "0000-0001-5218-3046"
+  - name: "Chris Hartgerink"
+    orcid: "0000-0003-1050-6809"
+categories: [R, open-source, package development, DOI]
+format:
+  html: 
+    toc: true
+# doi: "10.59350/z78kb-qrz59"
+---
+
+Licenses are an important topic within open source. Without licenses, information or code can be publicly available but not legally available for reuse or redistribution. The open source software community's most common licenses are [the MIT license](https://mit-license.org) or the [GNU GPLv3](https://www.gnu.org/licenses/gpl-3.0.html).
+
+When you read the MIT or GNU license, you can see they are rather specific:
+
+> Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the “Software”) [MIT License]
+
+and
+
+> The GNU General Public License is a free, copyleft license for software and other kinds of works.
+
+They aim to cover primarily software, not other forms of information such as, for example, data.
+
+### Licensing only code in R packages
+
+Given the importance of a license for redistribution, CRAN requires R packages to contain an [open-source license](https://cran.r-project.org/doc/FAQ/R-exts.html#Licensing).[^1] As one of the most important distributors for R packages historically, this rule has become the de facto standard.
+
+An R package is licensed appropriately when the license is stated in both the metadata `DESCRIPTION` file and the `LICENSE` file. The `LICENSE` file should include the license year and exact copyright holder (for example, an organization or a specific individual). Some R package developers choose to include a copy of the license but this is [not bundled in the R package when it is built](https://r-pkgs.org/license.html#key-files). You can [find an example of a `DESCRIPTION` file here](https://github.com/epiverse-trace/cfr/blob/29ee12aa9b8a49eedb9207d95c25c13d7d4e0ace/DESCRIPTION#L27) and [find an example `LICENSE` here](https://github.com/epiverse-trace/cfr/blob/main/LICENSE).
```

Comment:
This is a nitpick and maybe not worth the complication but: not all licenses require a `LICENSE` file. If the package is licensed GPL for example, just stating it in the `DESCRIPTION` is enough.

`LICENSE` is required for MIT and non-standard licenses

Reply:
```suggestion
An R package is licensed appropriately when the license is stated in the metadata `DESCRIPTION` file and, if necessary, in the `LICENSE` file. We recommend including the `LICENSE` file, with the license year and exact copyright holder (for example, an organization or a specific individual). Some R package developers choose to include a copy of the license but this is [not bundled in the R package when it is built](https://r-pkgs.org/license.html#key-files). You can [find an example of a `DESCRIPTION` file here](https://github.com/epiverse-trace/cfr/blob/29ee12aa9b8a49eedb9207d95c25c13d7d4e0ace/DESCRIPTION#L27) and [find an example `LICENSE` here](https://github.com/epiverse-trace/cfr/blob/main/LICENSE).
```

Like this maybe?